UI/window-basic-main-dropfiles.cpp

```diff
@@ -105,6 +110,12 @@ void OBSBasic::AddDropSource(const char *data, DropType image)
 		name = QUrl::fromLocalFile(QString(data)).fileName();
 		type = "browser_source";
 		break;
+	case DropType_Source:
+		json_settings = obs_data_create_from_json_file(data);
+		settings = obs_data_get_obj(json_settings, "settings");
+		name = obs_data_get_string(json_settings, "name");
+		type = obs_data_get_string(json_settings, "id");
+		break;
 	}
 
 	if (!obs_source_get_display_name(type))
```

Comment:
When I used obs_data_release like this:
```c
	case DropType_Source:
		json_settings = obs_data_create_from_json_file(data);
		settings = obs_data_get_obj(json_settings, "settings");
		name = obs_data_get_string(json_settings, "name");
		type = obs_data_get_string(json_settings, "id");
		obs_data_release(json_settings);
		break;
	}
```
the import would fail because the name value of the string was being released...if this'd cause a memory leak, there's already one in here it seems.

Reply:
Yes, there already is a memory leak (`settings` leaks at that same point); not asking you to fix that, just pointing out that you shouldn't introduce another memory leak (also, you can either fix this by replacing `obs_data_t json_settings = nullptr;` with `OBSData json_settings;` and adding `obs_data_release(json_settings);` right after `json_settings = obs_data_create_from_json_file(data);`, or you can use something like
```
	if (!obs_source_get_display_name(type)) {
		obs_data_release(json_settings);
		return;
	}
```

Reply:
Thanks for the catch, it didn't cross my mind about the `return;` at that point. I wrapped that part in an if block and properly released both pointers, so this'll solve the memory leak for the other one as well.